low-level/sensors/impl/centos_7/service_monitor.py

```diff
@@ -116,440 +298,139 @@ def initialize(self, conf_reader, msgQlist, product):
-                            self.services_to_monitor.remove(service)
-
-                    # Check for services in intermidiate state(not active)
-                    self.check_notactive_services()
-
-
-            logger.info("ServiceMonitor gracefully breaking out " +\
-                                "of dbus Loop, not restarting.")
+                iterations += 1
+            logger.info("ServiceMonitor gracefully breaking out " +
+                        "of dbus Loop, not restarting.")
         except GLib.Error as err:
             raise ThreadException(self.SENSOR_NAME,
-                "Ungrecefully breaking out of GLib.MainLoop() with error: %s"
-                %err)
+                                  "Ungrecefully breaking out of"
+                                  "GLib.MainLoop() with error: %s"
+                                  % err)
         except DBusException as err:
             raise ThreadException(self.SENSOR_NAME,
-                "Ungracefully breaking out of dbus loop with error: %s"% err)
+                                  "Ungracefully breaking out of dbus loop"
+                                  "with error: %s" % err)
         except Exception as err:
             raise ThreadException(self.SENSOR_NAME,
-                "Ungracefully breaking out of ServiceMonitor:run() "\
-                "with error: %s" % err)
-
-    def current_time(self):
-        """Returns the time as integer number in seconds since the epoch in UTC."""
-        return int(time.time())
-
-    def get_service_status(self, service = None, unit = None):
-        """Returns tuple of unit, service name, state, substate and pid."""
-        if not unit:
-            unit = self._bus.get_object('org.freedesktop.systemd1',\
-                                    self._manager.LoadUnit(service))
+                                  "Ungracefully breaking out of"
+                                  "ServiceMonitor:run() with error: %s" % err)
 
-        Iunit = Interface(unit, dbus_interface='org.freedesktop.DBus.Properties')
-
-        if not service:
-            service = str(Iunit.Get('org.freedesktop.systemd1.Unit', 'Id'))
-
-        state = str(Iunit.Get('org.freedesktop.systemd1.Unit', 'ActiveState'))
-        substate = str(Iunit.Get('org.freedesktop.systemd1.Unit', 'SubState'))
-        pid = str(Iunit.Get('org.freedesktop.systemd1.Service', 'ExecMainPID'))
-
-        return (unit, service, state, substate, pid)
+    def initialize_service(self, service_name):
+        try:
+            unit = self._bus.get_object(SYSTEMD_BUS,
+                                        self._manager.LoadUnit(service_name))
+            if Service.cache_exists(service_name):
+                service = Service.from_cache(service_name, unit)
+            else:
+                service = Service(unit)
+            self.services[service_name] = service
+            if service.enabled_on_disk:
```

Comment:
what enabled_on_disk means really?

Reply:
enabled_on_disk is flag which is true if service is enabled, false otherwise.